app/views/admin.py

```diff
@@ -93,3 +96,11 @@ def all_categories():
         cache_timeout=0,
         last_modified=datetime.now(),
     )
+
+
+@admin_blueprint.route("/<int:shop_id>/reset", methods=["GET"])
+def reset_conf(shop_id):
+    shop = Shop.query.get(shop_id)
+    reset(shop_id)
+    flash('Configurations was reseted !')
+    return redirect(url_for('admin.admin', shop_id=shop_id, shop=shop.name, **request.args))
```

Comment:
Here! If you call ```redirect``` - ```flash``` will not work... 
Or use ```render_template``` or remove ```flash```.
I recommend you the first one...